server/controllers/request.js

```diff
@@ -0,0 +1,95 @@
+const Request = require("../models/Request");
+const asyncHandler = require("express-async-handler");
+const {
+  updateRequest,
+  checkQueryForEmptyFields,
+  validateQueryForNewRequest,
+} = require("../utils/helperFunctions");
+
+// @route GET /request/load
+// @desc Get list of requests for logged-in user
+// @access Private
+exports.loadRequests = asyncHandler(async (req, res, next) => {
+  const uid = req.user.id;
+  let requests;
+
+  if (uid) {
+    requests = await Request.find({
+      userId: uid,
+    });
+  }
```

Comment:
So logically, this route will have the protect middleware. In that case, `req.user.id` will never be undefined and if the request doesn't have the auth cookie it will never hit this controller so we can clean this up a lot. We can just query for the users requests and store that in a constant then return a successful result since even empty arrays denote just no requests.

Hope this makes sense 👍🏻

Reply:
Sure, I see what you're saying. Makes sense! I actually grappled with that decision at first too, but decided to add the extra check in case, for example, the authentication were to go through a change, unwittingly exposing the route. Ultimately not a big deal either way on this one I think, but I wanted to explain the thought process here anyway.